File: src/main/java/dev/vrba/secrethitler/original/election/PresidentialAbility.java
package dev.vrba.secrethitler.original.election;

import org.jetbrains.annotations.NotNull;

import java.util.Map;

public enum PresidentialAbility {
    PEEK_TOP_THREE_POLICIES,
    INVESTIGATE_LOYALTY,
    CALL_SPECIAL_ELECTION,
    EXECUTE_PLAYER;

    public static @NotNull Map<Integer, PresidentialAbility> getAbilities(int players) {
        return switch (players) {
            case 5, 6 -> Map.of(
                    3, PEEK_TOP_THREE_POLICIES,
                    4, EXECUTE_PLAYER,
                    5, EXECUTE_PLAYER
            );

            case 7, 8 -> Map.of(
                    2, INVESTIGATE_LOYALTY,
                    3, CALL_SPECIAL_ELECTION,
                    4, EXECUTE_PLAYER,
                    5, EXECUTE_PLAYER
            );

            case 9, 10 -> Map.of(
                    1, INVESTIGATE_LOYALTY,
                    2, INVESTIGATE_LOYALTY,
                    3, CALL_SPECIAL_ELECTION,
                    4, EXECUTE_PLAYER,
                    5, EXECUTE_PLAYER
            );

            default -> throw new IllegalArgumentException("Number of players must be between 5 and 10 inclusive!");
        };
    }
}

File: src/main/kotlin/dev/vrba/secrethitler/engine/PresidentialAbility.kt
package dev.vrba.secrethitler.engine

fun getPresidentialAbilities(players: Int): Map<Int, PresidentialAbility> =
    when (players) {
        5, 6 -> mapOf(
            3 to PresidentialAbility.PeekPolicies,
            4 to PresidentialAbility.ExecutePlayer,
            5 to PresidentialAbility.ExecutePlayer,
        )

        7, 8 -> mapOf(
            2 to PresidentialAbility.InvestigatePlayerLoyalty,
            3 to PresidentialAbility.CallSpecialElection,
            4 to PresidentialAbility.ExecutePlayer,
            5 to PresidentialAbility.ExecutePlayer,
        )

        9, 10 -> mapOf(
            1 to PresidentialAbility.InvestigatePlayerLoyalty,
            2 to PresidentialAbility.InvestigatePlayerLoyalty,
            3 to PresidentialAbility.CallSpecialElection,
            4 to PresidentialAbility.ExecutePlayer,
            5 to PresidentialAbility.ExecutePlayer,
        )

        else -> throw IllegalArgumentException("Number of players must be between 5 and 10 inclusive.")
    }

enum class PresidentialAbility {
    PeekPolicies,
    InvestigatePlayerLoyalty,
    CallSpecialElection,
    ExecutePlayer
}